src/gui/dialogues/newpilotdialog.cpp

```diff
@@ -167,18 +167,20 @@ void NewPilotDialog::submitForm()
     newData.insert("displayname",displayName);
 
     using namespace experimental;
-    auto uin = newPilotInput(newData);
+    //auto uin = newPilotInput(newData);
+    auto entry = PilotEntry(newData);
+
+    if (role == Db::editExisting)
+        //entry.setPosition = oldEntry.position; //needs implementation in Entry class
 
-    switch (role) {
-    case Db::createNew:
-        DEB("New Object: " << newData);
+    if (!DB::commit(entry))
+    {
         /// [George]: we should check if db operation was succesful
         /// if not i assume we should just emit inputRejected or smth?
-        if(!DB::insert(uin)) emit QDialog::rejected();
-        break;
-    case Db::editExisting:
-        DEB("updating entry with: " << newData);
-        if(!DB::update(uin)) emit QDialog::rejected();
-        break;
+        /// [F] We should do something, emitting reject might not be the best
+        /// option since it closes the dialog. Maybe showing a QMessageBox with
+        /// what has gone wrong?
```

Comment:
Sure that is straight forward.
if `commit` returns false then **nothing** has been (should not be) altered and we simple re `submitForm` after prompting the error.

Reply:
We could also handle a signal. DB could have for example these signals:
```cpp
...
signals:
    void commitSuccessful();
    void commitUnsuccessful(QString errorMsg);
...
```
then we could actually show what has gone wrong instead of just having a ```false``` return code.

Reply:
Id like to study signals a bit more. I think lets keep the bool return for now because its very straight forward. If we use signals we propably have to handle them in the asyncronous way Qt handles them. We propably have to connect the Db signals with the dialog to provide a "reaction".

Reply:
```c++
class DB {
signals:
    void commitSuccessful();
    void commitUnsuccessful(QString errorMsg);
};

class NewPilotDialog {
slots:
    void onSuccessulCommit();
    void onUnsuccessfulCommit(QString msg);
    
    NewPilotDialog(...) {
        connect(&DB, &DB::commitSuccessful, this, &onSuccessfulCommit);
};
```
Im not sure if this will work because our database is static. Ill check it out once we have setup the dataflow. Its secondary atm. Just a different way of handling the same thing (success/unsuccess) on the db

Reply:
I think i figured out how to do it but it may require the DB to be a singleton. It isnt toomuch of a hassle but it needs some work. Once we deal with this merge ill work on it.